panel/src/components/Views/Pages/PageChangesView.vue

```diff
@@ -0,0 +1,180 @@
+<template>
+	<k-panel class="k-panel-inside k-page-changes-view">
+		<header class="k-page-changes-header">
+			<k-button-group>
+				<k-button
+					:link="link"
+					:responsive="true"
+					:text="$t('back')"
+					icon="angle-left"
+					size="sm"
+					variant="filled"
+				/>
+				<k-button
+					:icon="modes[mode].icon"
+					:dropdown="true"
+					:responsive="true"
+					:text="$t('view')"
+					size="sm"
+					variant="filled"
+					@click="$refs.layout.toggle()"
+				/>
+				<k-dropdown-content ref="layout" :options="dropdown" />
+			</k-button-group>
+
+			<k-form-controls
+				:editor="editor"
+				:is-locked="isLocked"
+				:is-unsaved="isUnsaved"
+				:modified="modified"
+				@discard="onDiscard"
+				@submit="onSubmit"
+			/>
+		</header>
+		<main class="k-page-changes-grid" :data-mode="mode">
+			<section v-if="mode === 'latest' || mode === 'compare'">
+				<k-headline>
+					{{ modes.latest.label }}
+					<k-button
+						:link="src.latest"
+						icon="open"
+						size="xs"
+						target="_blank"
+						variant="filled"
+					/>
+				</k-headline>
+				<iframe :src="src.latest"></iframe>
+			</section>
+			<section v-if="mode === 'changes' || mode === 'compare'">
+				<k-headline>
+					{{ modes.changes.label }}
+					<k-button
+						v-if="isUnsaved"
+						:link="src.changes"
+						icon="open"
+						size="xs"
+						target="_blank"
+						variant="filled"
+					/>
+				</k-headline>
+				<iframe v-if="isUnsaved" :src="src.changes"></iframe>
+				<k-empty v-else>{{ $t("lock.unsaved.empty") }}</k-empty>
+			</section>
+		</main>
+	</k-panel>
+</template>
+
+<script>
+import PageView from "@/components/Views/Pages/PageView.vue";
+
+export default {
+	extends: PageView,
+	props: {
+		src: Object
+	},
+	data() {
+		return {
+			mode: "compare"
+		};
+	},
+	computed: {
+		modes() {
+			return {
+				latest: {
+					label: this.$t("version.latest"),
+					icon: "layout-right",
+					current: this.mode === "latest",
+					click: () => (this.mode = "latest")
+				},
+				changes: {
+					label: this.$t("version.changes"),
+					icon: "layout-left",
+					current: this.mode === "changes",
+					click: () => (this.mode = "changes")
+				},
+				compare: {
+					label: this.$t("version.compare"),
+					icon: "layout-columns",
+					current: this.mode === "compare",
+					click: () => (this.mode = "compare")
+				}
+			};
+		},
+		dropdown() {
+			return [this.modes.latest, this.modes.changes, "-", this.modes.compare];
+		}
+	},
+	methods: {
+		async onDiscard() {
+			if (this.isLocked === true) {
+				return false;
+			}
+
+			await this.$panel.content.discard();
+			await this.$panel.view.open(this.link);
```

Comment:
`Thoughts:` I didn't find it the most pleasant UX to be thrown out of the preview just because of saving/discarding changes. Was a bit like "woa, what's happening, where am I?". So I was thinking if that's necessary or we could just remain on the preview view and letting the user decide what to do next? On the other hand there is not much to preview without any changes, so I also do understand the thought of adding it. What's your feeling with it? Does it feel good that with the button press also the view dissolves?

Reply:
Hm, I really never thought about keeping people in the view after saving/discarding. It would be indeed a pretty useless place to stay to be honest.